packages/build-standalone/src/plugins/esbuild-measure-file-size.ts

```diff
@@ -0,0 +1,95 @@
+import recursive from 'recursive-readdir';
+import path from 'path';
+import * as fs from 'fs-extra';
+import { sync as gzipSize } from 'gzip-size';
+
+import type { ModularBuildConfig, ModularBuildContext } from '../types';
+import { canReadAsset, removeFileNameHash } from '../utils';
+import type { WorkflowPlugin } from '../workflow';
+
+export const FILE_SIZES = Symbol('FILE_SIZES');
```

Comment:
Why a `Symbol`? The only way of accessing this field for another plugin is to depend on this file and this seems intentional - shouldplugins be coupled like this?

Reply:
For this particular example, the result of

```
const previousFileSizes = await esbuildMeasureFileSizesBeforeBuild( paths.appBuild );
```

Is only used in:

```
printFileSizesAfterBuild(assets, previousFileSizes);
```

So it makes sense that the data only be accessible to the `esbuildMeasureFileSizes` plugin... note that this plugin hooks into the build lifecycle at two points - the `validate` and `cleanup` phases before and after the build itself.

I would like the plugins to be agnostic of each other, so where possible they would be storing data in the context using a Symbol. Where they really do need to share state, the context should probably have a better way of storing state than the `Record<string, unknown>` type it has at the moment, perhaps by:

- exporting the symbols from the plugins to be used elsewhere
- have a few named properties on the context itself (eg `{ fileSizes?: FileSizes } & Record<string, unknown>`)
- use the plugin factories as keys for the data in a map:

```
function makeWorkflow(config) {
  return createWorkflow(
    doesSomething(config),
    usesSomething(config, doesSomething),
  );
}

function doesSomething(config) {
  return {
    phase: 'validate',
    handler (context) {
      context.get(doesSomething, { /* data */ });
    }
  };
}

function usesSomething(config, doesSomething) {
  return {
    phase: 'cleanup',
    handler (context) {
      const something = context.get(doesSomething);
    }
  };
}
```